compiler/resolution/lowerIterators.cpp

```diff
@@ -139,6 +139,24 @@ static void nonLeaderParCheck()
   USR_STOP();
 }
 
+static bool isVirtualIterator(Symbol* iterator) {
+  Vec<Type*> children = iterator->type->dispatchChildren;
+  return !((children.n == 0) || (children.n == 1 && children.v[0] == dtObject));
+}
+
+static void parallelIterVirtualCheck() {
+  forv_Vec(BlockStmt, block, gBlockStmts) {
+    if (isAlive(block) && block->isForLoop()) {
+      ForLoop* forLoop = toForLoop(block);
+      Symbol* iterator = toSymExpr(forLoop->iteratorGet())->symbol();
+      FnSymbol* ifn = getTheIteratorFn(iterator);
+      if (ifn->hasFlag(FLAG_INLINE_ITERATOR) && isVirtualIterator(iterator)) {
```

Comment:
A comment saying that FLAG_INLINE_ITERATOR implies that the iterator is parallel or something like that would be helpful here.

Reply:
I have a branch that I plan on merging post-release that changes `FLAG_INLINE_ITERATOR` to `FLAG_PARALLEL_ITERATOR`.